tests/functional/base_test.go

```diff
@@ -94,37 +96,66 @@ func CreateOVNDBCluster(namespace string, OVNDBClusterName string, spec map[stri
 	return th.CreateUnstructured(raw)
 }
 
+func UpdateOVNDBCluster(cluster *ovnv1.OVNDBCluster) {
+	k8sClient.Update(ctx, cluster)
+}
+
 func OVNDBClusterConditionGetter(name types.NamespacedName) condition.Conditions {
 	instance := GetOVNDBCluster(name)
 	return instance.Status.Conditions
 }
 
+func ScaleDBCluster(name types.NamespacedName, replicas int32) {
+	Eventually(func(g Gomega) {
+		c := GetOVNDBCluster(name)
+		*c.Spec.Replicas = replicas
+		g.Expect(k8sClient.Update(ctx, c)).Should(Succeed())
+	}).Should(Succeed())
+}
+
 // CreateOVNDBClusters Creates NB and SB OVNDBClusters
-func CreateOVNDBClusters(namespace string, nad string) []types.NamespacedName {
+func CreateOVNDBClusters(namespace string, nad map[string][]string, replicas int32) []types.NamespacedName {
 	dbs := []types.NamespacedName{}
 	for _, db := range []string{v1beta1.NBDBType, v1beta1.SBDBType} {
 		name := fmt.Sprintf("ovn-%s", uuid.New().String())
 		spec := GetDefaultOVNDBClusterSpec()
+		stringNad := ""
+		// OVNDBCluster doesn't allow multiple NADs, hence map len
+		// must be <= 1
+		Expect(len(nad)).Should(BeNumerically("<=", 1))
+		for k, _ := range nad {
+			if strings.Contains(k, "/") {
```

Comment:
and if it doesn't contain it, then what? you silently allow to create the cluster without attaching it to NAD? I think it's better to fail explicitly here so that test author is aware this is happening.

Reply:
Done